### ISSUE TITLE
Occasionally getting error "SnapshotTesting/UIImage.swift:24: Fatal error: Unexpectedly found nil while unwrapping an Optional value"

### ISSUE THREAD

**Issue by simondelphia** [pointfreeco/swift-snapshot-testing#613]
Occasionally getting an error on CI

> SnapshotTesting/UIImage.swift:24: Fatal error: Unexpectedly found nil while unwrapping an Optional value

This is referring to a line here:
https://github.com/pointfreeco/swift-snapshot-testing/blob/main/Sources/SnapshotTesting/Snapshotting/UIImage.swift

Which has some force unwraps.

I'm not sure why this is happening — possibly not even related to this repo.

But instead of force unwrapping and leading to a fatal error, this could presumably be handled more gracefully instead.

**Comment by niemmi-gs** [comment]
Hello @simondelphia,
I got the same error, please check if *.png files are ok, in my case my was corrupted.

**Comment by simondelphia** [comment]
Yes they are corrupted. The cause of the issue has nothing to do with this library, so I didn't get into the details. Just would be nice if the library failed more gracefully.

**Comment by nhaarman** [comment]
@simondelphia But do you know how you resolved this issue?

**Comment by KamilGucik** [comment]
Check if you have lfs configured. In my case it was the reason of the problem. I had to install them on CI

**Comment by simonbromberg** [comment]
Yes iirc this was a case of something going wrong with git lfs. Point is however that the library ought not to force unwrap like this and instead should fail gracefully in this case.

I don't remember exactly what we did to fix it. I know at one point we were trying to set up Xcode Cloud and we might have seen this because the test bundle didn't actually have access to the bundled files without some special workarounds we had to implement. But I'm sure we also saw this in other cases where there was just some problem with loading a specific image occasionally.

**Comment by ZevEisenberg** [comment]
It would be great if, failing to decode the image, it would fall back to decoding a String as UTF-8, just to see if there's actually text data in there. That would save a lot of confusion when the reason the data is not decoding is because it's actually the placeholder LFS file because your LFS config is wonky.
